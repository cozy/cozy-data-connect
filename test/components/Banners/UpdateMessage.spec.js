'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { tMock } from '../../jestLib/I18n'
import { UpdateMessage } from 'components/Banners/UpdateMessage'

describe('UpdateMessage component', () => {
  it(`Should be render correctly if not blocking update`, () => {
    const component = shallow(<UpdateMessage t={tMock} />)
      .shallow()
      .dive()
      .getElement()
    expect(component).toMatchSnapshot()
  })

  it(`Should be render correctly if blocking update`, () => {
    const component = shallow(<UpdateMessage t={tMock} isBlocking />)
      .shallow()
      .dive()
      .getElement()
    expect(component).toMatchSnapshot()
  })
})
