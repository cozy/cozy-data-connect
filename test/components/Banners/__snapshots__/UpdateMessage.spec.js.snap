// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateMessage component Should be render correctly if blocking update 1`] = `
<div
  className="Infos Infos--danger u-maw-none"
>
  <Stack
    spacing="m"
    tag="div"
  >
    <div
      className="Infos-description"
    >
      <Stack
        spacing="s"
        tag="div"
      >
        <React.Fragment>
          <SubTitle
            className="u-pomegranate"
            ellipsis={false}
            tag="div"
          >
            An update is available for this service.
          </SubTitle>
          <div
            className="u-flex u-w-100"
          >
            <Text
              className="u-pl-half"
              ellipsis={false}
              tag="div"
            >
              Update it to keep fetching your data
            </Text>
          </div>
        </React.Fragment>
      </Stack>
    </div>
    <div>
      <DefaultButton
        className="u-m-0"
        disabled={false}
        label="Update"
        onClick={[Function]}
        theme="danger"
      />
    </div>
  </Stack>
</div>
`;

exports[`UpdateMessage component Should be render correctly if not blocking update 1`] = `
<div
  className="Infos Infos--secondary u-maw-none"
>
  <Stack
    spacing="m"
    tag="div"
  >
    <div
      className="Infos-description"
    >
      <Stack
        spacing="s"
        tag="div"
      >
        <React.Fragment>
          <SubTitle
            className=""
            ellipsis={false}
            tag="div"
          >
            An update is available for this service.
          </SubTitle>
          <div
            className="u-flex u-w-100"
          >
            <Text
              className="u-pl-half"
              ellipsis={false}
              tag="div"
            >
              Perform this update to keep fetching your data and to have the latest features
            </Text>
          </div>
        </React.Fragment>
      </Stack>
    </div>
    <div>
      <DefaultButton
        className="u-m-0"
        disabled={false}
        label="Update"
        onClick={[Function]}
        theme="danger"
      />
    </div>
  </Stack>
</div>
`;
